kmir/src/kmir/prove.py

```diff
@@ -0,0 +1,52 @@
+import logging
+import sys
+from pathlib import Path
+from typing import Final
+
+from .kmir import KMIR
+
+_LOGGER: Final = logging.getLogger(__name__)
+_LOG_FORMAT: Final = '%(levelname)s %(asctime)s %(name)s - %(message)s'
+
+
+def prove(
+    kmir: KMIR,
+    spec_file: Path,
+    *,
+    # bug_report: BugReport | None = None, handled at init KMIR
+    save_directory: Path | None = None,
+    reinit: bool = False,
+    depth: int | None = None,
+    smt_timeout: int | None = None,
+    smt_retry_limit: int | None = None,
+    trace_rewrites: bool = False,
+    # kore_rpc_command: str | Iterable[str] | None = None,
+) -> None:
+    _LOGGER.info('Extracting claims from file')
+
+    if kmir.prover:
+        kmir_prover = kmir.prover
+    else:
+        raise ValueError('The prover object in kmir is not initialised.')
+
+    claims = kmir_prover.get_all_claims(spec_file)
+    assert not claims, ValueError(f'No claims found in file {spec_file}')
```

Comment:
```suggestion
    if not claims:
        raise ValueError(f'No claims found in file {spec_file}')
```

Reply:
@tothtamas28 I've fixed this. The error I am asking for help was rather a silly mistake where the cli expects `spec_file` while `exe_prove` takes `spec-file`.

all good now.